Modules/_tkinter.c

```diff
@@ -3022,17 +3040,89 @@ Tkapp_WantObjects(PyObject *self, PyObject *args)
 /*[clinic input]
 _tkinter.tkapp.willdispatch
 
+Skips main loop wait polling until next call of mainloop finishes
+
+DEPRECATED
+
+call(), var_invoke(), and createcommand() will wait indefinitely for main loop
+Sets the internal dispatching flag regardless of mainloop dispatch state
+May lead to unexpected hangs or inconsistencies in dispatching
+
 [clinic start generated code]*/
 
 static PyObject *
 _tkinter_tkapp_willdispatch_impl(TkappObject *self)
-/*[clinic end generated code: output=0e3f46d244642155 input=d88f5970843d6dab]*/
+/*[clinic end generated code: output=0e3f46d244642155 input=7b46376275304a4a]*/
 {
+    if (PyErr_WarnEx(PyExc_DeprecationWarning,
+        "willdispatch() is deprecated; in the future it will have no effect.", 1))
```

Comment:
So its not being removed? Not saying this is bad at all, but I would appreciate hearing your reasoning.

Reply:
Right, with the future behavior of waiting indefinitely, the only change necessary for deprecation will be to keep this method from corrupting the return value of `dispatching`. So we could remove it or not, but leaving it as a null op would break less code. I suppose it's a point we should bring up with Serhiy and/or Guilherme when the time comes.